### PR TITLE
Fixed an issue where route refresh was not updating the ETA in RouteProgress

### PR DESCRIPTION
**Pull request by LukasPaczos**
### Description

Nav Native since version `v90` (available in `v2.4.0-beta.2` and later) expects us to pass a full refresh response back to `updateAnnotations` API, instead of the previous collections of annotations. This bug prevents us from updating the ETA values returned in `RoutesProgress` while the congestion on the route line (for example) is correctly updated. This can be noticed on the latest version of the Nav SDK by this log:
```
E/Mapbox: [nav-native] TrackingAnnotationsUpdater::update json parsing error: [json.exception.out_of_range.403] key 'route' not found
```

Since we have access to the full refresh response only in the internal `RouterWrapper`, and the public `NavigationRouterRefreshCallback` which can be used with a custom router instance works only with `NavigationRoute` references, we won't be able to access the full response reliably in the rest of the SDK.

To mitigate this problem, I'm adding somewhat of a workaround that manufactures the refresh response out of the annotations found in the refreshed route object right before passing it to Nav Native. We can reconsider the interfaces and this setup with the next major version of the SDK.

Additionally I'm improving our refresh instrumentation test to wait for a signal from Nav Native that the annotations were updated successfully in the native codebase as well.

### CODE REVIEW

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt [hunk 1]
```diff
@@ -108,7 +160,7 @@ class RouteRefreshTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.ja
 
         assertEquals(sanityDuration, initialDuration, 0.0)
         assertEquals(227.918, initialDuration, 0.0001)
-        assertEquals(230.651, refreshedDuration, 0.0001)
+        assertEquals(1180.651, refreshedDuration, 0.0001)
```
Comment (VysotskiVadim) [inline instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt:163]: should we also check duration from route progress to make sure that NN got actual data?

Comment (LukasPaczos) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt:163]: We do that above.

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt [hunk 2]
```diff
@@ -98,6 +125,31 @@ class RouteRefreshTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.ja
...
+        progressIdlingResource.register()
+        Espresso.onIdle()
+        progressIdlingResource.unregister()
+
+        val latch = CountDownLatch(1)
+        runOnMainSync {
+            mapboxNavigation.registerRouteProgressObserver { routeProgress ->
+                val durationRemaining = routeProgress.durationRemaining
+                val expectedDurationRemaining = 1180.651
+                // 30 seconds margin of error
```
Comment (RingerJK) [inline instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt:141]: why is it 30 seconds? Should it be the exact value as we use mockk data?

Comment (LukasPaczos) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt:141]: I'm not sure. This is an integration test after all, I assume native navigator could introduce a margin of error depending on how the location we provide is snapped to the route.

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt (hunk 1 above)
Comment (RingerJK) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt:163]: yeah, actually If we want to cover as many cases and catch potential regression, we should compare all possible change data from NN

File: libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigatorImpl.kt [hunk 3]
```diff
@@ -189,21 +194,49 @@ object MapboxNativeNavigatorImpl : MapboxNativeNavigator {
...
-     * @param route [DirectionsRoute]
+     * This methods manufactures a [DirectionsRefreshResponse] to adhere to requirements from
+     * https://github.com/mapbox/mapbox-navigation-native/pull/5420 where the full response has to be provided
+     * to [Navigator.updateAnnotations], not only the annotations/incidents collections.
      */
     override suspend fun updateAnnotations(route: NavigationRoute) {
-        route.directionsRoute.legs()?.forEachIndexed { index, routeLeg ->
+        val refreshedLegs = route.directionsRoute.legs()?.map { routeLeg ->
+            RouteLegRefresh.builder()
+                .annotation(routeLeg.annotation())
```
Comment (SiarheiFedartsou) [inline libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigatorImpl.kt:204]: Is it possible to add incidents here too?

**Review by SiarheiFedartsou — APPROVED** [review]

File: libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigatorImpl.kt (hunk 3 above)
Comment (RingerJK) [reply libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigatorImpl.kt:204]: yes, it is. Can re-reuse the `udpateAnnotation` API for incidents? (not intuitive actually)

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt (hunk 1 above)
Comment (LukasPaczos) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt:163]: That should be tested on the Nav Native side. I'm just checking that roundtripping works.

**Review by RingerJK — APPROVED** [review]

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt (hunk 1 above)
Comment (VysotskiVadim) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt:167]: > We do that above.

As far as I see we check durations from legs which are set by Android SDK, not NN 🤔

Comment (LukasPaczos) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt:167]: We check the duration from route progress in https://github.com/mapbox/mapbox-navigation-android/pull/5755/files#diff-bef8d61571a3abff3c6dfcd44a23faa32209dcd71521bdd29e50bd3d13fc7e95R146.

**Pull request merged**